homeassistant/components/light/iglo.py

```diff
@@ -0,0 +1,124 @@
+"""
+Support for lights under the iGlo brand.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.iglo/
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import (CONF_HOST, CONF_NAME, CONF_PORT)
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, ATTR_RGB_COLOR, ATTR_COLOR_TEMP,
+    SUPPORT_BRIGHTNESS, SUPPORT_COLOR_TEMP, SUPPORT_RGB_COLOR,
+    Light, PLATFORM_SCHEMA
+)
+
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['iglo==1.0.0']
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = 'iGlo Light'
+DEFAULT_PORT = 8080
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the iGlo lighs."""
+    host = config.get(CONF_HOST)
+    name = config.get(CONF_NAME)
+    port = config.get(CONF_PORT)
+    add_devices([IGloLamp(name, host, port)], True)
+
+
+class IGloLamp(Light):
+    """Representation of an iGlo light."""
+
+    def __init__(self, name, host, port):
+        """Initialize the light."""
+        from iglo import Lamp
+        self._name = name
+        self._lamp = Lamp(0, host, port)
+        self._on = True
+        self._brightness = 255
+        self._rgb = (0, 0, 0)
+        self._color_temp = 0
+
+    @property
+    def name(self):
+        """Return the name of the light."""
+        return self._name
+
+    @property
+    def brightness(self):
+        """Return the brightness of this light between 0..255."""
+        return int((self._brightness / 200.0) * 255)
+
+    @property
+    def color_temp(self):
+        """Return the color temperature."""
+        return self._color_temp
+
+    @property
+    def min_mireds(self):
+        """Return the coldest color_temp that this light supports."""
+        return 1
```

Comment:
I just saw this merged so I know I am too late to the party.

Anyway, this is supposed to be in _mired_, the unknown "reciprocal megakelvin" unit.

You find the mired value with this formula: 1000000/kelvin. So if the light does 7000 Kelvin as its coldest white, the min_mired value should be 143.

Same goes for max_mireds and color_temp.

Reply:
I don't really understand the mirad thing, but the lights accept a colour temp value between 1 and 255 to change between warm and cool white. Maybe I will look into it more once I move back into the part of my house with my new lights and start using this on my main HA.

Reply:
Doing it this way will mean that examples for other lights do not work properly. It also breaks the `kelvin` alias for white temperature.

But now you know where the problem is, once you notice :)